src/Command/App/AppInitCommand.php

```diff
@@ -0,0 +1,244 @@
+            },
+        ]);
+
+        return $fields;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        // The project root is a Git repository, as we assume there are no
+        // config files yet.
+        $projectRoot = $this->findTopDirectoryContaining('.git');
+
+        /** @var QuestionHelper $questionHelper */
+        $questionHelper = $this->getHelper('question');
+        $options = $this->form->resolveOptions($input, $output, $questionHelper);
+
+        $configFile = self::$config->get('service.app_config_file');
+
+        $configFileAbsolute = isset($options['directory'])
+            ? sprintf('%s/%s/%s', $projectRoot, $options['directory'], $configFile)
+            : sprintf('%s/%s', $projectRoot, $configFile);
+
+        if (file_exists($configFileAbsolute) && !$questionHelper->confirm('The config file already exists. Overwrite?')) {
+            return 1;
+        }
+
+        $this->makeAppYaml($configFileAbsolute, $options);
+        $this->makeRoutingYaml($projectRoot, $options['name']);
+        $this->makeServicesYaml($projectRoot);
+
+        return 0;
+    }
+
+    /**
+     * Generates the .platform.app.yaml file, based on the user-supplied information.
+     *
+     * @param string $configFile
+     *   The absolute path to the file to create.
+     * @param array $appConfig
+     *   The user-supplied configuration information from which to generate a file.
+     */
+    protected function makeAppYaml($configFile, $appConfig)
+    {
+        $this->stdErr->writeln('Creating config file: ' . $configFile);
+
+        unset($appConfig['directory'], $appConfig['subdir']);
+
+        (new Filesystem())->dumpFile($configFile, Yaml::dump($appConfig, 10));
+    }
+
+    /**
+     * Generates a stock routing.yaml file.
+     *
+     * @param string $projectRoot
+     *   The absolute path to the project root.
+     * @param string $applicationName
+     *   The user-supplied name of the application.
+     */
+    protected function makeRoutingYaml($projectRoot, $applicationName)
+    {
+        $this->stdErr->writeln('Creating routing file.');
+
+        mkdir($projectRoot . '/.platform');
+
+        $routingYaml = <<<END
+# The routing.yaml file describes how an incoming URL is going
+# to be processed by Platform.sh.  With the defaults below, all requests to
+# The the domain name configured in the UI will pass through to the application
+# and all requests to the www. prefix will be redirected to the bare domain.
+# To reverse that behavior, simply swap the definitions.
+#
+# See https://docs.platform.sh/user_guide/reference/routes-yaml.html for more information.
+
+"http://{default}/":
+  type: upstream
+  upstream: "{$applicationName}:http"
+"http://www.{default}/":
+  type: redirect
+  to: "http://{default}/"
+
+END;
+
+        (new Filesystem())->dumpFile($projectRoot . '/.platform/routing.yaml', $routingYaml);
```

Comment:
D'Oh. This is what peer review is for. :smile: (routing.yml is what Drupal calls its files, so that's what's burned into my brain.)